realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/admin/ServerAdmin.kt

```diff
@@ -80,10 +81,18 @@ class ServerAdmin {
         result = JSONObject(executeRequest(builder))
         groupId = (result.getJSONArray("roles")[0] as JSONObject).getString("group_id")
 
-        // Get Internal App Id
+        // Get Internal App Id of the requested app
+        val appId = this.app.configuration.appId
         builder = Request.Builder().url("$baseUrl/groups/$groupId/apps").get()
-        result = JSONArray(executeRequest(builder))[0] as JSONObject
-        appId = result.getString("_id")
+        val response = JSONArray(executeRequest(builder))
```

Comment:
Interesting and good catch 🎉 . I wonder why it even worked in the first place before? And why it was only flaky?

Reply:
Because with two apps the first one is sometimes the right one 😄 Maybe they are sorted by generated ids o.a.

Reply:
Doh, ofcourse 👍

Reply:
The tricky parts is why it was almost only happening on CI?

Reply:
Yup, I guess it is a timing issue that just surfaces there. Also, we would need to understand what the server uses for sorting when returning those apps. Maybe it is just based on how the data is stored in MongoDB instead of a specific sorting.